packages/pwa/app/commerce-api/auth.js

```diff
@@ -236,18 +229,18 @@ class Auth {
             throw new HTTPError(response.status, json.message)
         }
 
-        const tokenBody = createGetTokenBody(
-            response.url,
-            `${getAppOrigin()}${slasCallbackEndpoint}`,
-            window.sessionStorage.getItem('codeVerifier')
-        )
+        let tokenURL = response.url
+        if (!this._onClient) {
+            tokenURL = response.headers.get('location')
+        }
+
+        const tokenBody = createGetTokenBody(tokenURL, redirect_uri, codeVerifier)
```

Comment:
This is a bit weird: server side, fetch options are respected and the redirect is not followed.

Client side, this is not the case and the redirect is followed.